drivechain-server/api/wallet/wallet.go

```diff
@@ -287,24 +286,29 @@ func (s *Server) ListSidechainDeposits(ctx context.Context, c *connect.Request[p
 
 // CreateSidechainDeposit implements walletv1connect.WalletServiceHandler.
 func (s *Server) CreateSidechainDeposit(ctx context.Context, c *connect.Request[pb.CreateSidechainDepositRequest]) (*connect.Response[pb.CreateSidechainDepositResponse], error) {
+	// TODO: this is an external address. Would be nice to be possible to
+	// fetch an internal address
+	changeAddress, _, err := s.wallet.GetNewAddress(ctx)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
I think this should be inside the fundTx-function. This assumes we're always going to have change, but that's not the case.

We're also going to generate too many addresses, if it's before validation of the arguments. We ønly want to generate an address when it's actually needed.